overleaf/notifications
# [misc] migrate to the native mongo driver

### Description

For https://github.com/overleaf/issues/issues/2907
This PR is migrating the mongo driver from mongojs to mongodb -- the native driver.

The current revision is a bit unclean as it mixes Promises with callbacks. Refactoring the service to async/await is too much work to code and review at this time.

#### Related Issues / PRs

https://github.com/overleaf/issues/issues/2907
Previous migration: https://github.com/overleaf/chat/pull/53 https://github.com/overleaf/chat/pull/54 https://github.com/overleaf/docstore/pull/66 (https://github.com/overleaf/contacts/pull/49 https://github.com/overleaf/document-updater/pull/144 https://github.com/overleaf/github-sync/pull/57 pending in review)

### Review

The healtch check are accessing the db directly. I split the PR into the app-part and health-check part to ease the verification of this PR -- e.g. run health-check on the app-commit then on both.

<details>
<summary> changed startup </summary>

The startup behavior has been changed in the chat service already https://github.com/overleaf/chat/pull/54 and is running with success in prod by now.

> This PR changes the startup of the service to not accept any http requests until after the mongo connection has been established. It also adds the correct way to handle connection setup errors, to catch and log them -- vs just letting them bubble up as unhandled PromiseRejections.
> 
> The web service can start without an established/working mongo connection, which is bad. Eventually we can port this snippet over.

</details>

#### Potential Impact

High. All interactions which persist changes to mongo.

#### Manual Testing Performed

Run the health-check and see no artifacts in the db after running it. Run the find-codemod -- no changes.

- create two accounts Alice and Bob
- Alice creates a new Project and invites Bob as collaborator
- Bob sees the invite on the project dashboard
- Bob can accept the invite -- soft-delete in mongo (`templateKey` field removed)
- Log out Bob; Clear site-data in the browser (localStorage); Log back in as Bob; See no project-invite notification
